### PR TITLE
[WIP] Add Dressipi RecSys Util Function to Merlin Models

### PR DESCRIPTION
**Pull request by bschifferer**
Similar to ALICCP, we added util function for dressipi recsys dataset + a synthetic dataset schema file

### CODE REVIEW

File: merlin/datasets/synthetic.py [hunk 1]
```diff
@@ -70,6 +71,7 @@ def generate_data(
     - criteo
     - aliccp
     - aliccp-raw
+    - drepessi
```
Comment (rnyak) [inline merlin/datasets/synthetic.py:74]: I guess theres is a typo here.

File: merlin/datasets/ecommerce/__init__.py [hunk 2]
```diff
@@ -20,5 +20,12 @@
...
 )
+from merlin.datasets.ecommerce.dressipi.dataset import get_drepessi2022
 
-__all__ = ["prepare_alliccp", "transform_aliccp", "get_aliccp", "default_aliccp_transformation"]
+__all__ = [
+    "prepare_alliccp",
+    "transform_aliccp",
+    "get_aliccp",
+    "default_aliccp_transformation",
+    "get_drepessi2022",
```
Comment (rnyak) [inline merlin/datasets/ecommerce/__init__.py:30]: `get_drepessi2022` there is a typo in this func name.

File: merlin/datasets/ecommerce/dressipi/dataset.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+def get_drepessi2022(input_path):
+    """
+    Load the dressipi dataset from the RecSys 2022 Challenge.
+
+    We can't download this dataset automatically so this needs to be downloaded manually
+    To use this function, prepare the data by following these 3 steps:
+    1. Sign up and download the data from
+    [dressipi-recsys2022.com](https://www.dressipi-recsys2022.com/).
+    2. Unzip the raw data to a directory.
+    3. Run `get_drepessi2022(data_dir)` to convert the data to Merlin datasets
```
Comment (rnyak) [inline merlin/datasets/ecommerce/dressipi/dataset.py:61]: Better to add an an error msg that pops up when someone wants to use get_drepessi2022  func without downloading the dataset.

File: merlin/datasets/synthetic.py (hunk 1 above)
Comment (rvk007) [reply merlin/datasets/synthetic.py]: Fixed the typo.

File: merlin/datasets/ecommerce/__init__.py (hunk 2 above)
Comment (rvk007) [reply merlin/datasets/ecommerce/__init__.py]: Fixed the typo.

**Review by bschifferer — COMMENTED** [review]
Can we rename test_usercase_... to test_usecase_.. without the r

File: examples/usecases/ecommerce-session-based-next-item-prediction-for-fashion.ipynb [hunk 4]
```diff
@@ -0,0 +1,1546 @@
+{
```
Comment (bschifferer) [inline examples/usecases/ecommerce-session-based-next-item-prediction-for-fashion.ipynb:1]: <p>can we remove the import for cudf. we will delete the reference later as well -&gt; <code>import cudf</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA-Merlin/models/pull/664/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (bschifferer) [inline examples/usecases/ecommerce-session-based-next-item-prediction-for-fashion.ipynb:1]: <p>lets remove dataset = nvt.Dataset(cudf.concat([train.to_ddf().compute(), valid.to_ddf().compute()]))</p><p> </p><p>and we use workflow.fit(train)</p><p> </p><p>Lets add a description above.</p><p> </p><p>"We call fit and transform similar to the scikit learn API.</p><p> </p><p>Categorify will map unkown item_ids (and purchase_ids) to a special category <code>0</code> in the validation dataset. This can bias the validation metrics. In our example, almost all item_ids in validation are available in train and we neglect it. </p><p>"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA-Merlin/models/pull/664/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='29'/>

Comment (bschifferer) [inline examples/usecases/ecommerce-session-based-next-item-prediction-for-fashion.ipynb:1]: <p>*loaded -&gt; same tense</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA-Merlin/models/pull/664/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='40'/>

Comment (bschifferer) [inline examples/usecases/ecommerce-session-based-next-item-prediction-for-fashion.ipynb:1]: <p>It doent average the sequencial input features? Otherwise, we couldnt use a BiLSTM.</p><p> </p><p>The text should be smth like:</p><p>"which concatenates the embedding vectors for all sequential features (item_id_list, f_47_list, f_68_list) per step (e.g. here 3)". The concatenated vectors are processed by a BiLSTM. The hidden state of the BiLSTM is concatenated with the embedding vectors of the categorical features"...</p><p> </p><p>"Then we connect it with a Multi-Layer Perceptron Block." -&gt; there is no sequential afterwards anymore</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA-Merlin/models/pull/664/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='58'/>

Comment (bschifferer) [inline examples/usecases/ecommerce-session-based-next-item-prediction-for-fashion.ipynb:1]: <p>Now we will a Bi-LSTM model by defining a custom layer of type mm.Block. It takes the sequential interacrion embeddings as an input</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA-Merlin/models/pull/664/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='63'/>

Comment (rnyak) [inline examples/usecases/ecommerce-session-based-next-item-prediction-for-fashion.ipynb:1]: <p>I would rephrase this sentence "We leveraged them for our RecSys2022 participations" as "We leveraged them for our RecSys practitioners."</p><p> </p><p>I find this sentence not clear <code>When the probabilities predicted by a Classification model are higher than its accuracy</code>. accuracy of the model do you mean? or number of samples classified correctly?</p><p> </p><p>I think this should be fixed as:</p><p> </p><p>α: Label smoothing hyper-parameter ( 0 &lt;= α &lt;= 1)</p><p> </p><p> </p><p>This sentence does not read well " we share the Embedding layer's weights which is used to convert the input to embeddings, as the softmax weights,  to convert hidden layer output to softmax layer output." . Please rephrase it, you can express it in 2 sentences to make it clear.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA-Merlin/models/pull/664/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (rnyak) [inline examples/usecases/ecommerce-session-based-next-item-prediction-for-fashion.ipynb:1]: <p>May be remove the dots from end of each sentence in the bullets above, only keep the last one.</p><p> </p><p>shall we add any explanation (1-2 sentences) how we decided those 22 features are the most important features?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA-Merlin/models/pull/664/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (rnyak) [inline examples/usecases/ecommerce-session-based-next-item-prediction-for-fashion.ipynb:1]: <p>replace <code>continuous integers</code> with <code><strong>contiguous integers</strong></code><strong>. </strong>  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA-Merlin/models/pull/664/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Comment (rnyak) [inline examples/usecases/ecommerce-session-based-next-item-prediction-for-fashion.ipynb:1]: <p>"The NVTabular Op <code>GroupBy</code> enables the transformation" rephrase as "The NVTabular <code>GroupBy</code> op enables the transformation."</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA-Merlin/models/pull/664/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Comment (rnyak) [inline examples/usecases/ecommerce-session-based-next-item-prediction-for-fashion.ipynb:1]: <p>I guess better to say <code>We will Tag the columns accordingly based on the type of each column.</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA-Merlin/models/pull/664/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>

Comment (rnyak) [inline examples/usecases/ecommerce-session-based-next-item-prediction-for-fashion.ipynb:1]: <p>Nice plot!</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA-Merlin/models/pull/664/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='35'/>

Comment (bschifferer) [inline examples/usecases/ecommerce-session-based-next-item-prediction-for-fashion.ipynb:1]: <p>Now, we combine <code>Bi-LSTM</code> block with <code>InputBlock</code> of categorical features by concatenating them</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA-Merlin/models/pull/664/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='67'/>

Comment (bschifferer) [inline examples/usecases/ecommerce-session-based-next-item-prediction-for-fashion.ipynb:1]: <p>I am not sure, if that is ParallelBlock. Maybe we just say that we build the Model by chaining concats with mlp_block and prediction_Task</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA-Merlin/models/pull/664/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='73'/>

Comment (rnyak) [reply examples/usecases/ecommerce-session-based-next-item-prediction-for-fashion.ipynb:1]: <p>I am a bit confused with your response Benedikt: <code>Otherwise, we couldnt use a BiLSTM.</code> I assume you have a simple MLP first, and then below you use BiLSTM.  why cant you concat the sequential lists with averaging if this is a simple MLP model?</p>

File: examples/usecases/ecommerce-session-based-next-item-prediction-for-fashion.ipynb [hunk 5]
```diff
@@ -0,0 +1,1640 @@
...
```
Comment (rnyak) [inline examples/usecases/ecommerce-session-based-next-item-prediction-for-fashion.ipynb:1]: <p>What does <code>we neglect it</code> mean? what did we neglect? the items that are in validation set but not in train set? if yes, please write down clearly. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA-Merlin/models/pull/664/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='28'/>

Comment (rnyak) [inline examples/usecases/ecommerce-session-based-next-item-prediction-for-fashion.ipynb:1]: <p>Is this really working as intended without setting <code>seq=True</code> and  <code>max_seq_length=3</code> in the InputBlock? </p><p> </p><p>https://github.com/NVIDIA-Merlin/models/blob/main/merlin/models/tf/inputs/base.py#L47-L48</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA-Merlin/models/pull/664/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='47'/>

Comment (bschifferer) [reply examples/usecases/ecommerce-session-based-next-item-prediction-for-fashion.ipynb:1]: In the `seq_inputs ` we are setting the parameters seq=True and max_seq_length=3.

In the `cat_inputs `, we do not need them as this is a simple categorical feature, no sequence

**Comment by bschifferer** [comment]
rerun tests

**Review by rnyak — APPROVED** [review]
